cli/pcluster/easyconfig.py

```diff
@@ -244,11 +245,41 @@ def configure(args):  # noqa: C901 FIXME!!!
         if e.errno != errno.EEXIST:
             raise  # can safely ignore EEXISTS for this purpose...
 
+    if not _is_config_valid(args, config):
+        sys.exit(1)
+
+    # If we are here, than the file it's correct and we can override it.
     # Write configuration to disk
     open(config_file, "a").close()
     os.chmod(config_file, stat.S_IRUSR | stat.S_IWUSR)
     with open(config_file, "w") as cf:
         config.write(cf)
 
+
+def _is_config_valid(args, config):
+    """
+    Validate the configuration of the pcluster configure.
+    :param args: the arguments passed with the command line
+    :param config: the configParser
+    :return True if the configuration is valid, false otherwise
+    """
+    # We create a temp_file to validate before overriding the original config
+    path = os.path.join(tempfile.gettempdir(), "temp_config")
+    temp_file = path
+    temp_args = args
```

Comment:
Actually there is a bug with that code because of the lack of defensive copy, which means that if the user has selected a path that would be overwritten